src/ray/raylet/scheduling/scheduling_policy.cc

```diff
@@ -14,41 +14,76 @@
 
 #include "ray/raylet/scheduling/scheduling_policy.h"
 
+#include <functional>
+
 namespace ray {
 
 namespace raylet_scheduling_policy {
+namespace {
+bool IsCPUOnlyRequest(const ResourceRequest &resource_request) {
+  if (resource_request.predefined_resources.size() <= GPU) {
+    return true;
+  }
+  return resource_request.predefined_resources[GPU] == 0;
+}
+bool DoesNodeHaveGPUs(const NodeResources &resources) {
+  if (resources.predefined_resources.size() <= GPU) {
+    return false;
+  }
+  return resources.predefined_resources[GPU].total > 0;
+}
+}  // namespace
 
-int64_t HybridPolicy(const ResourceRequest &resource_request, const int64_t local_node_id,
-                     const absl::flat_hash_map<int64_t, Node> &nodes,
-                     float spread_threshold, bool force_spillback,
-                     bool require_available) {
+int64_t HybridPolicyWithFilter(const ResourceRequest &resource_request,
```

Comment:
@stephanie-wang @wuisawesome could you guys confirm if this is the right high-level approach?

As I understand, this policy is the central place for node prioritization, correct? I.e., it has the final say on where lease requests are granted.